lib/index.js

```diff
@@ -40,6 +40,18 @@ const endMeasure = (baseName) => {
   performance.clearMarks(`${baseName}-start`);
 };
 
+const ccBinaryFound = (execPath, err) => {
+  if (err.code === 'ENOENT' && !badPaths.has(execPath)) {
+    const msg = 'codeclimate binary not found! Installation ' +
+      'instructions at http://github.com/codeclimate/codeclimate';
+    atom.notifications.addError(msg);
+    // Only notify once per path
+    badPaths.add(execPath);
+    return false;
+  }
+  return true;
```

Comment:
Why not to leverage the Atom notifications API in this case? Right now, since the shell call has been dropped, the error will be always related to the CodeClimate CLI. Instead my `ccBinaryFound()` function we could use next generic function:

```javascript
const notifyError = (cmd, err, msg, buttons = []) => {
  let friendlyDesc = '';
  switch (err.code) {
    case 'ENOENT':
      friendlyDesc = 'CodeClimate binary could not be found.'; break;
    case 'EACCES':
    case 'EDIR':
      friendlyDesc = 'Executable path not pointing to a binary.'; break;
    default: friendlyDesc = 'CodeClimate execution failed.';
  }
  if (!badCommands.has(cmd)) {
    atom.notifications.addError('linter-codeclimate error', {
      buttons: [{
        className: 'btn-install',
        onDidClick: () => {
          require('shell').openExternal('https://github.com/codeclimate/codeclimate');
        },
        text: 'Install guide',
      }].concat(buttons),
      detail: `${msg || friendlyDesc}
Exception details:
- CODE: ${err.code}
- MESSAGE: ${err.message}
- COMMAND: \`${cmd}\`
`,
      dismissable: true,
      stack: err.stack,
    });
    // Only notify once per path
    badCommands.add(cmd);
  }
};
```

which shows an error notification with the error code/message, the codeclimate command executed, an explanation and the stack trace, all without opening the dev console. Can be customized through the `switch()` and the optional parameters (message and buttons).

Changing `badPaths` to `badCommands` and using it to cache all the executed commands instead all the executable paths, the user can get a notification for each error only once, and in case of unhandled errors, you will always get the same info as before, but the generic message `CodeClimate execution failed`.

Here's how it looks:

![linter-codeclimate](https://cloud.githubusercontent.com/assets/10707639/25019435/d22daae4-208a-11e7-8c5c-23c83170b761.png)